File: examples/supplychain-app/fabric/chaincode_rest_server/rest-server/src/models/ProductRequest.ts
import { deserialize } from 'class-transformer';
import { IsDefined, IsNotEmpty, IsUUID } from 'class-validator';
import { Validateable } from 'mixins/Validateable';

export class ProductRequest extends Validateable {
  static fromJSON(json: string): ProductRequest {
    return deserialize(ProductRequest, json);
  }

  @IsNotEmpty()
  @IsDefined()
  productName: string;

  @IsNotEmpty()
  @IsDefined()
  health: string;

  @IsNotEmpty()
  @IsDefined()
  misc: { [key: string]: any };

  @IsNotEmpty()
  @IsDefined()
  @IsUUID()
  trackingID: string;

  @IsNotEmpty()
  @IsDefined()
  counterparties: string[];

  lastScannedAt: string;
}

File: examples/supplychain-app/fabric/chaincode_rest_server/chaincode/common/ProductRequest.go
package common

// The ProductRequest models request body for new product in a supply chain
type ProductRequest struct {
	ID           string                 `json:"trackingID"`
	ProductName  string                 `json:"productName"`
	Health       string                 `json:"health"`
	Metadata     map[string]interface{} `json:"misc"`
	Location     string                 `json:"lastScannedAt"`
	Participants []string               `json:"counterparties"`
}
